hi-ml-azure/src/health_azure/himl.py

```diff
@@ -685,67 +648,72 @@ def _str_to_path(s: Optional[PathOrString]) -> Optional[Path]:
     return s
 
 
-def create_v2_inputs(ml_client: MLClient, input_datasets: List[DatasetConfig]) -> Dict[str, Input]:
-    """
-    Create a dictionary of Azure ML v2 Input objects, required for passing input data in to an AML job
+def get_data_assets_from_configs(ml_client: MLClient, dataset_list: List[DatasetConfig]) -> List[Data]:
```

Comment:
I think you can write the function `create_v2_outputs` (and .. inputs) more easily if you refactor this method to take a single `DatasetConfig` and outputs a single `DAta` object. The downstream functions will then simply be dictionary comprehensions.